cmdstanpy/model.py

```diff
@@ -949,21 +949,33 @@ def sample(
             if (
                 force_one_process_per_chain is None
                 and not cmdstan_version_before(2, 28, info_dict)
+                and stan_threads is not None
                 and stan_threads == 'true'
             ):
                 one_process_per_chain = False
                 num_threads = parallel_chains * num_threads
                 parallel_procs = 1
-            elif (
-                force_one_process_per_chain is False
-                and cmdstan_version_before(2, 28, info_dict)
-            ):
-                get_logger().warning(
-                    'Installed version of CmdStan cannot multi-process chains, '
-                    'will run %d processes. '
-                    'Run "install_cmdstan" to upgrade to latest version.',
-                    chains,
-                )
+            if force_one_process_per_chain is False:
+                if cmdstan_version_before(2, 28, info_dict):
+                    get_logger().warning(
+                        'Installed version of CmdStan cannot multi-process '
+                        'chains, will run %d processes. '
+                        'Run "install_cmdstan" to upgrade to latest version.',
+                        chains,
+                    )
+                elif stan_threads is None or stan_threads == 'false':
+                    get_logger().warning(
+                        'Stan program not compiled for threading, '
+                        'will run %d processes. '
+                        'Recompile model and specify argument '
+                        '"cpp_options={\'STAN_THREADS\':\'TRUE\'}.',
+                        chains,
+                    )
```

Comment:
I don't think we should do this IMO. CmdStan provides the functionality to do multiple chains in one process without threading, we should let users have that if they explicitly ask for it. We can emit a messaging saying that is what is happening, but we shouldn't make that feature impossible to access